src/services/web3/bnb-contract.service.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@nestjs/common';
+
+import { ethers } from 'ethers';
+import { binanceProvider } from 'config/system';
+
+
+@Injectable()
+export class BnbContractService {
+
+  protected readonly provider: ethers.providers.JsonRpcProvider;
+  protected readonly contract: ethers.Contract;
+
+  constructor(address: string, abi: any) {
+    this.provider = new ethers.providers.JsonRpcProvider(binanceProvider);
```

Comment:
For initialize better use OnModuleInit

Reply:
What if I need to pass arguments to the constructor? Is it possible with OnModuleInit?

Reply:
Done.